pkg/helm/watcher/controller/helm_watcher.go

```diff
@@ -75,16 +88,22 @@ func (r *HelmWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	values := make(cache.ValueMap)
 
 	for _, chart := range charts {
+		if v, err := r.checkForNewVersion(ctx, chart); err != nil {
+			log.Error(err, "checking for new versions failed")
+		} else if v != "" {
+			log.Info("sending notification event for new version", "version", v)
+			r.sendEvent(log, &repository, repository.Status.Artifact.Revision, "info", chart.Name, v)
```

Comment:
some duplication here, passing in `repository` and `repository.Status.Artifact.Revision`? could this be condensed to just `repository`?